### PR TITLE
[SPARK-36007][INFRA] Failed to run benchmark in GA

### PR DESCRIPTION
**Pull request by pingsutw**
### What changes were proposed in this pull request?

When I'm running the benchmark in GA, I met the below error.

https://github.com/pingsutw/spark/runs/2867617238?check_suite_focus=true 
```
java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.j
ava:1692)java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
21/06/20 07:40:02 ERROR SparkContext: Error initializing SparkContext.java.lang.AssertionError: assertion failed: 
spark.test.home is not set! at scala.Predef$.assert(Predef.scala:223) at org.apache.spark.deploy.worker.Worker.<init>
(Worker.scala:148) at org.apache.spark.deploy.worker.Worker$.startRpcEnvAndEndpoint(Worker.scala:954) at 
org.apache.spark.deploy.LocalSparkCluster.$anonfun$start$2(LocalSparkCluster.scala:68) at 
org.apache.spark.deploy.LocalSparkCluster.$anonfun$start$2$adapted(LocalSparkCluster.scala:65) at 
scala.collection.immutable.Range.foreach(Range.scala:158) at 
org.apache.spark.deploy.LocalSparkCluster.start(LocalSparkCluster.scala:65) at 
org.apache.spark.SparkContext$.org$apache$spark$SparkContext$$createTaskScheduler(SparkContext.scala:2954) at 
org.apache.spark.SparkContext.<init>(SparkContext.scala:559) at org.apache.spark.SparkContext.<init>
(SparkContext.scala:137) at 
org.apache.spark.serializer.KryoSerializerBenchmark$.createSparkContext(KryoSerializerBenchmark.scala:86) at 
org.apache.spark.serializer.KryoSerializerBenchmark$.sc$lzycompute$1(KryoSerializerBenchmark.scala:58) at 
org.apache.spark.serializer.KryoSerializerBenchmark$.sc$1(KryoSerializerBenchmark.scala:58) at 
org.apache.spark.serializer.KryoSerializerBenchmark$.$anonfun$run$3(KryoSerializerBenchmark.scala:63)
```

### Why are the changes needed?

Set `spark.test.home` in the benchmark workflow.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Rerun the benchmark in my fork.
https://github.com/pingsutw/spark/actions/runs/996067851

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -85,6 +85,7 @@ jobs:
         # In benchmark, we use local as master so set driver memory only. Note that GitHub Actions has 7 GB memory limit.
         bin/spark-submit \
```
Comment (HyukjinKwon) [inline .github/workflows/benchmark.yml:86]: I think it's better to set the `SPARK_HOME` env instead. Can you add:

```
SPARK_HOME: ${{ github.workspace }}
```

under `env` with a comment that explains why we should set it?

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
Looks good otherwise

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (pingsutw) [reply .github/workflows/benchmark.yml]: Thanks for the review. Updated it and retrigger the benchmark workflow in my fork.
https://github.com/pingsutw/spark/actions/runs/997492502

**Review by HyukjinKwon — APPROVED** [review]

**Comment by pingsutw** [comment]
Benchmark workflow has run successfully.
https://github.com/pingsutw/spark/actions/runs/997616866

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
I merged to branch-3.2 too

**Pull request closed without merging**